### ISSUE TITLE
dnsmasq LaunchDaemon plist is installed with wrong ownership

### ISSUE THREAD

**Issue by haakonstorm** [Homebrew/brew#6873]
* [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
* [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
* [x] ran `brew update` and can still reproduce the problem?
* [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
* [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
* [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

# TL;DR

Running `sudo brew services start dnsmasq` - the daemon property list should be added with `root:wheel`, but is added with `root:admin` permissions. This causes problems.

## general

- As the original problem originated on a production server running latest Mojave, pardon my ignorance but the logs and details under here are from my workstation mac, running macOS 10.15 Catalina beta 5. 
- There are the usual pre-release warnings, but results are the same as with the Mojave server, reproducible as far as I can see, and the relevant logs and debug is provided below.

## brew install dnsmasq

```
==> Downloading https://homebrew.bintray.com/bottles/dnsmasq-2.80.mojave.bottle.1.tar.gz
==> Downloading from https://akamai.bintray.com/4d/4d150c19c5c856435f9d38307c4b9fb153f942
######################################################################## 100.0%
==> Pouring dnsmasq-2.80.mojave.bottle.1.tar.gz
==> Caveats
To have launchd start dnsmasq now and restart at startup:
  sudo brew services start dnsmasq
==> Summary
🍺  /usr/local/Cellar/dnsmasq/2.80: 8 files, 531.3KB
```
### sudo brew services start dnsmasq

```
==> Tapping homebrew/services
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-services'...
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 14 (delta 0), reused 8 (delta 0), pack-reused 0
Unpacking objects: 100% (14/14), done.
Tapped 1 command (43 files, 59.5KB).
Warning: Taking root:admin ownership of some dnsmasq paths:
  /usr/local/Cellar/dnsmasq/2.80/sbin
  /usr/local/Cellar/dnsmasq/2.80/sbin/dnsmasq
  /usr/local/opt/dnsmasq
  /usr/local/opt/dnsmasq/sbin
  /usr/local/var/homebrew/linked/dnsmasq
This will require manual removal of these paths using `sudo rm` on
brew upgrade/reinstall/uninstall.
==> Successfully started `dnsmasq` (label: homebrew.mxcl.dnsmasq)
```

### What you expected to happen

- `dnsmasq` LaunchDaemon `.plist` correctly installed  in `/Library/LaunchDaemons`

### What happened (permissions from my workstation)
- `.plist`installed but with wrong posix group permissions 
- (in relation to valid plist deployment in /Library/LaunchDaemons domain):
<pre><code>
/Library/LaunchDaemons
❯ ll
total 88
-rw-r--r--  1 root  wheel   727B 16 jul 19:59 xx.xxxxx.xxxxxxxxxxxxx.xxxxx
-rw-r--r--  1 root  wheel   564B 28 jul 20:56 xxx.xxxxx.xxx.xxxxxxxxx.xx.xxxxx
-rw-r--r--  1 root  wheel   777B 12 aug 18:15 xxx.xxxxxxxxx.xxxxxxxxxxxxxxxxx.xxxxx
-rw-r--r--  1 root  wheel   899B  7 aug 21:23 xxx.xxxxxx.xxxxxxxxxxx.xxxxx.xxxxx
-rw-r--r--  1 root  wheel   232B 15 aug 04:12 xxx.xxxxxxxxx.xxxxxxxxxx.xxxxxx.xxxxx
-rw-r--r--  1 root  wheel   657B  9 jun 22:39 xxx.xxxxxxxxx.xxxxxx.xxxx.xxxxxx.xxxxx
-rw-r--r--  1 root  wheel   596B 19 jul 22:37 xxx.xxxx-xxxx.xxxxxxxxxx.xxxxxx.xxxxx
<b>-rw-r--r--  1 root  <i>admin </i>  576B 15 aug 13:13 homebrew.mxcl.dnsmasq.plist</b>
-rw-r--r--  1 root  wheel   573B 14 aug 17:20 xxx.xxxx.xxxxxxxxx.xxxxx.xxxxx
-rw-r--r--  1 root  wheel   509B 14 aug 17:20 xxx.xxxx.xxxxx.xxxxxxx.xxxxx
-rw-r--r--  1 root  wheel   515B 14 aug 17:20 xxx.xxxx.xxxxxxxxx.xxxxxx.xxxxx    
</code></pre>

(Emphasis added for clarity)

## From the production server

<pre><code>
➜  LaunchDaemons ls -la
total 80
drwxr-xr-x  12 root  wheel   384 15 aug 09:14 .
drwxr-xr-x+ 63 root  wheel  2016 12 jul 17:40 ..
-rw-r--r--   1 root  wheel   590 30 jun 22:41 xxxxxxxxxxxxxxxx-xxxxxxxxxxxxxx.xxxxx
-rw-r--r--   1 root  wheel   632 26 jul 03:56 xxx.xxxxx.xxxxxxxxx.xxxxxxxxxxxxxxxx.xxxxx
-rw-r--r--   1 root  wheel   559 12 jul 17:40 xxx.xxxxx.xxxxxxx.xxxxx
-rw-r--r--   1 root  wheel   484 17 jul  2017 xxx.xxxxxxx.xxxxxxxxx.xxxxx 
-rw-r--r--   1 root  wheel   701 17 jul  2017 xxx.xxxxxxx.xxxxxxxxxxx.xxxxx
-rw-r--r--   1 root  wheel   488 17 jul  2017 xxx.xxxxxxx.xxxxxx.xxxxx
<b>-rw-r--r--   1 root  <i>wheel</i>   723 17 jul  2017 xxx.xxxxxxx.xxxxxxxxxxx.xxxxx </b>
-rw-r--r--   1 root  wheel   596 30 jul 23:24 xxx.xxxx-xxxx.xxxxxxxxxxxxx.xxxxxx.xxxxx
-rw-r--r--   1 root  admin   576 15 aug 09:14 homebrew.mxcl.dnsmasq.plist
-rw-r--r--   1 root  wheel   382  3 jul 14:07 xxx.xxxxxxxxx.xxxxxxxx.xxxxx 
</code></pre>

## The daemon property list should be added with `root:wheel`, but is added with `root:admin` permissions.

*From [the documentation](https://developer.apple.com/library/archive/technotes/tn2083/_index.html#//apple_ref/doc/uid/DTS10003794-CH1-SUBSECTION43):*

```
When you install your daemon, make sure that you set the file system permissions correctly. Apple recommends that daemons be owned by root, have an owning group of wheel, and use permissions 755 (rwxr-xr-x) for executables and directories, and 644 (rw-r--r--) for files. In addition, every directory from your daemon up to the root directory must be owned by root and only writable by the owner (or owned by root and sticky). If you don't do this correctly, a non-admin user might be able to escalate their privileges by modifying your daemon (or shuffling it aside).
```

### brew gist-logs --verbose --debug (on Mojave production server)

```
➜  LaunchDaemons brew gist-logs dnsmasq --verbose --debug
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/dnsmasq.rb
Error: No logs.
/usr/local/Homebrew/Library/Homebrew/cmd/gist-logs.rb:124:in `load_logs'
/usr/local/Homebrew/Library/Homebrew/cmd/gist-logs.rb:41:in `gistify_logs'
/usr/local/Homebrew/Library/Homebrew/cmd/gist-logs.rb:152:in `gist_logs'
/usr/local/Homebrew/Library/Homebrew/brew.rb:102:in `<main>'
```

### brew gist-logs --verbose --debug (on my Catalina workstation)

```
❯ brew gist-logs dnsmasq --verbose --debug
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/dnsmasq.rb
Error: You are using macOS 10.15.
We do not provide support for this pre-release version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Discourse, Twitter or IRC. You are responsible for resolving any issues you
experience, as you are running this pre-release version.

Error: Kernel.exit
/usr/local/Homebrew/Library/Homebrew/install.rb:72:in `exit'
/usr/local/Homebrew/Library/Homebrew/install.rb:72:in `diagnostic_checks'
/usr/local/Homebrew/Library/Homebrew/install.rb:47:in `perform_preinstall_checks'
/usr/local/Homebrew/Library/Homebrew/cmd/gist-logs.rb:150:in `gist_logs'
/usr/local/Homebrew/Library/Homebrew/brew.rb:102:in `<main>'
```

- I haven't had the opportunity to test much more than this with the actual Mojave production server
- nonetheless the permissions for the plist are wrong according to the spec/documentation, as far as I can tell. 
- I might be wrong with all this, I know dnsmasq starts with elevated privileges and then drops down to another user (usually `nobody`). 
- Please advise if I understand the docs wrongly.

Thanks for all the great work with brew and all the formulae. Best wishes from Oslo.

**Comment by fxcoudert** [comment]
Generic brew issue, the ownership/group is not included in the formula itself.

**Comment by MikeMcQuaid** [comment]
We'll consider a pull request to https://github.com/homebrew/homebrew-services that changes the group but we do not state that we'll set the group to anything in particular and you're the first person I've heard raise this complaint.

**Comment by haakonstorm** [comment]
You're totally right, ownership isn't included in the formula, I wasn't aware, I would have filed an issue with homebrew-services directly. 

But since this is so - the logical conclusion then is that any *system* daemon installed with `brew services` is done using faulty permissions. 

Again, just to be clear - as stated by Apple in the docs:

> (...) Apple recommends that daemons be **owned by root**,  have an 
> **owning group of wheel**, and use permissions **755 (rwxr-xr-x) for executables** 
> and directories, and **644 (rw-r--r--) for files.**  (...)

I don't understand why they state "recommends" and not "requires" when it comes to daemons manifested in the `system` domain, as nothing will work as you'd expect otherwise. 

To be clear: you want me to open a PR with 'https://github.com/homebrew/homebrew-services'?

**Comment by MikeMcQuaid** [comment]
> To be clear: you want me to open a PR with 'https://github.com/homebrew/homebrew-services'?

Yes, please.

> I don't understand why they state "recommends" and not "requires" when it comes to daemons manifested in the `system` domain, as nothing will work as you'd expect otherwise.

Root services already work for many people (myself included).
